server/src/main/java/org/opensearch/cluster/service/ClusterService.java

```diff
@@ -108,6 +108,7 @@ public ClusterService(
         this.nodeName = Node.NODE_NAME_SETTING.get(settings);
         this.clusterManagerService = clusterManagerService;
         this.operationRouting = new OperationRouting(settings, clusterSettings);
+        this.operationRouting.setClusterService(this);
```

Comment:
why not modify the constructor itself ?

Reply:
The constructor is used at 16 places, mostly test files. So if we change the constructor, it will require us to change all the tests as well. I did this to avoid changes at multiple places.

Reply:
It should be okay the modify the test. Even they would need to have it set , else would have NPE,  right ?

Reply:
There won't be any NPE, since none of the older tests need clusterService

Reply:
Changed this. Injecting dependency now